src/main/java/net/kaczmarzyk/spring/data/jpa/utils/SpecificationBuilder.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+import net.kaczmarzyk.spring.data.jpa.web.ProcessingContext;
+import net.kaczmarzyk.spring.data.jpa.web.SpecificationFactory;
+import net.kaczmarzyk.spring.data.jpa.web.StandaloneProcessingContext;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Jakub Radlica
+ * @author Kacper Leśniak (Tratif sp. z o.o.)
+ * SpecificationBuilder allows creating specification apart from web layer.
+ * It is recommended to use builder methods that corresponding to the type of argument passed to sepcification.
+ * params = <args> => withParams(<argName>, <values...>), single param argument can provide multiple values
+ * pathVars = <args> => withPathVar(<argName>, <value>), single pathVar argument can provide single value
+ * headers = <args> => withHeader(<argName>, <value>), single header argument can provide single value
```

Comment:
1. please change order (@authors after description) -- see other javadocs for reference
2. Make sure that this doc is rendered as html as well (use <ul><li> for list), not only plaintext
3. (you can take a  look on how IDE is showing this javadoc)